src/Components/test/testassets/BasicTestApp/Program.cs

```diff
@@ -54,7 +53,36 @@ public static async Task Main(string[] args)
                 return new PrependMessageLoggerFactory("Custom logger", originalLogger);
             });
 
-            await builder.Build().RunAsync();
+            var host = builder.Build();
+            ConfigureCulture(host);
+
+            await host.RunAsync();
+        }
+
+        private static void ConfigureCulture(WebAssemblyHost host)
+        {
+            // In the absence of a specified value, we want the culture to be en-US so that the tests for bind can work consistently.
+            var culture = new CultureInfo("en-US");
+
+            Uri uri = null;
+            try
+            {
+                uri = new Uri(host.Services.GetService<NavigationManager>().Uri);
+            }
+            catch (ArgumentException)
+            {
+                // Some of our tests set this application up incorrectly so that querying NavigationManager.Uri throws.
```

Comment:
If the app isn't correctly hosted under the `BaseUri`, accessing `NavigationManager.Uri` throws. A bunch of the MapFallbackToClientSideBlazor have this issue.

Reply:
Are you seeing this before or after rebasing https://github.com/dotnet/aspnetcore/pull/20019?

The test failures in MapFallbackToClientSideBlazor appeared (and were resolved) in that PR. If you saw this issue before rebasing, something fishy might be afoot.

If you saw it after rebasing, it might be a race condition in the initialization of the NavigationManager...

Reply:
I saw this before I rebased on your changes. This is very likely an issue with one of the test apps.